{"nodes":[{"pos":[639,734],"content":"Represents the exception that is thrown when incorrectly trying to create or access a relation.","needQuote":true,"needEscape":true,"nodes":[{"content":"Represents the exception that is thrown when incorrectly trying to create or access a relation.","pos":[0,95]}]},{"pos":[747,1737],"content":"The InvalidConstraintException is thrown when incorrectly invoking the following methods while trying to create or access a relation.      -   <xref:System.Data.DataRelationCollection.Add%2A?displayProperty=fullName>      -   <xref:System.Data.DataRowCollection.Clear%2A?displayProperty=fullName>      -   <xref:System.Data.DataRow.GetParentRow%2A?displayProperty=fullName>       If you call <xref:System.Data.DataRelationCollection.Add%2A>, the InvalidConstraintException may be thrown if a relation cannot be created based on the values supplied in the parameters. If you call <xref:System.Data.DataRowCollection.Clear%2A>, the InvalidConstraintException may be thrown if a <xref:System.Data.ForeignKeyConstraint> is enforced on the <xref:System.Data.DataRowCollection>. If you call <xref:System.Data.DataRow.GetParentRow%2A>, the InvalidConstraintException may be thrown if the columns have different data types or the tables do not belong to the same <xref:System.Data.DataSet>.","needQuote":false,"needEscape":true,"nodes":[{"content":"The InvalidConstraintException is thrown when incorrectly invoking the following methods while trying to create or access a relation.      -   <xref:System.Data.DataRelationCollection.Add%2A?displayProperty=fullName>      -   <xref:System.Data.DataRowCollection.Clear%2A?displayProperty=fullName>      -   <xref:System.Data.DataRow.GetParentRow%2A?displayProperty=fullName>       If you call <xref:System.Data.DataRelationCollection.Add%2A>, the InvalidConstraintException may be thrown if a relation cannot be created based on the values supplied in the parameters. If you call <xref:System.Data.DataRowCollection.Clear%2A>, the InvalidConstraintException may be thrown if a <xref:System.Data.ForeignKeyConstraint> is enforced on the <xref:System.Data.DataRowCollection>. If you call <xref:System.Data.DataRow.GetParentRow%2A>, the InvalidConstraintException may be thrown if the columns have different data types or the tables do not belong to the same <xref:System.Data.DataSet>.","pos":[0,982],"nodes":[{"content":"The InvalidConstraintException is thrown when incorrectly invoking the following methods while trying to create or access a relation.","pos":[0,133]},{"content":"-   &lt;xref:System.Data.DataRelationCollection.Add%2A?displayProperty=fullName&gt;      -   &lt;xref:System.Data.DataRowCollection.Clear%2A?displayProperty=fullName&gt;      -   &lt;xref:System.Data.DataRow.GetParentRow%2A?displayProperty=fullName&gt;       If you call &lt;xref:System.Data.DataRelationCollection.Add%2A&gt;, the InvalidConstraintException may be thrown if a relation cannot be created based on the values supplied in the parameters.","pos":[139,566],"source":"      -   <xref:System.Data.DataRelationCollection.Add%2A?displayProperty=fullName>      -   <xref:System.Data.DataRowCollection.Clear%2A?displayProperty=fullName>      -   <xref:System.Data.DataRow.GetParentRow%2A?displayProperty=fullName>       If you call <xref:System.Data.DataRelationCollection.Add%2A>, the InvalidConstraintException may be thrown if a relation cannot be created based on the values supplied in the parameters."},{"content":"If you call &lt;xref:System.Data.DataRowCollection.Clear%2A&gt;, the InvalidConstraintException may be thrown if a &lt;xref:System.Data.ForeignKeyConstraint&gt; is enforced on the &lt;xref:System.Data.DataRowCollection&gt;.","pos":[567,772],"source":" If you call <xref:System.Data.DataRowCollection.Clear%2A>, the InvalidConstraintException may be thrown if a <xref:System.Data.ForeignKeyConstraint> is enforced on the <xref:System.Data.DataRowCollection>."},{"content":"If you call &lt;xref:System.Data.DataRow.GetParentRow%2A&gt;, the InvalidConstraintException may be thrown if the columns have different data types or the tables do not belong to the same &lt;xref:System.Data.DataSet&gt;.","pos":[773,982],"source":" If you call <xref:System.Data.DataRow.GetParentRow%2A>, the InvalidConstraintException may be thrown if the columns have different data types or the tables do not belong to the same <xref:System.Data.DataSet>."}]}]},{"pos":[2398,2498],"content":"Initializes a new instance of the <bpt id=\"p1\">&lt;xref href=\"System.Data.InvalidConstraintException\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class.","needQuote":true,"needEscape":true,"source":"Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class."},{"pos":[3135,3261],"content":"Initializes a new instance of the <bpt id=\"p1\">&lt;xref href=\"System.Data.InvalidConstraintException\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class with the specified string.","needQuote":true,"needEscape":true,"source":"Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class with the specified string."},{"pos":[3404,3455],"content":"The string to display when the exception is thrown.","needQuote":true,"needEscape":true,"nodes":[{"content":"The string to display when the exception is thrown.","pos":[0,51]}]},{"pos":[4254,4385],"content":"Initializes a new instance of the <bpt id=\"p1\">&lt;xref href=\"System.Data.InvalidConstraintException\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class with serialization information.","needQuote":true,"needEscape":true,"source":"Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class with serialization information."},{"pos":[4665,4729],"content":"The data that is required to serialize or deserialize an object.","needQuote":true,"needEscape":true,"nodes":[{"content":"The data that is required to serialize or deserialize an object.","pos":[0,64]}]},{"pos":[4825,4902],"content":"Description of the source and destination of the specified serialized stream.","needQuote":true,"needEscape":true,"nodes":[{"content":"Description of the source and destination of the specified serialized stream.","pos":[0,77]}]},{"pos":[5523,5729],"content":"Initializes a new instance of the <bpt id=\"p1\">&lt;xref href=\"System.Data.InvalidConstraintException\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class with a specified error message and a reference to the inner exception that is the cause of this exception.","needQuote":true,"needEscape":true,"source":"Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception."},{"pos":[5910,5971],"content":"The error message that explains the reason for the exception.","needQuote":true,"needEscape":true,"nodes":[{"content":"The error message that explains the reason for the exception.","pos":[0,61]}]},{"pos":[6045,6243],"content":"The exception that is the cause of the current exception, or a null reference (<bpt id=\"p1\">&lt;xref uid=\"langword_csharp_Nothing\" name=\"Nothing\" href=\"\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> in Visual Basic) if no inner exception is specified.","needQuote":true,"needEscape":true,"source":"The exception that is the cause of the current exception, or a null reference (<xref uid=\"langword_csharp_Nothing\" name=\"Nothing\" href=\"\"></xref> in Visual Basic) if no inner exception is specified."}],"content":"### YamlMime:ManagedReference\nitems:\n- uid: System.Data.InvalidConstraintException\n  id: InvalidConstraintException\n  children:\n  - System.Data.InvalidConstraintException.#ctor\n  - System.Data.InvalidConstraintException.#ctor(System.String)\n  - System.Data.InvalidConstraintException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)\n  - System.Data.InvalidConstraintException.#ctor(System.String,System.Exception)\n  langs:\n  - csharp\n  name: InvalidConstraintException\n  nameWithType: InvalidConstraintException\n  fullName: System.Data.InvalidConstraintException\n  type: Class\n  summary: Represents the exception that is thrown when incorrectly trying to create or access a relation.\n  remarks: \"The InvalidConstraintException is thrown when incorrectly invoking the following methods while trying to create or access a relation.  \\n  \\n-   <xref:System.Data.DataRelationCollection.Add%2A?displayProperty=fullName>  \\n  \\n-   <xref:System.Data.DataRowCollection.Clear%2A?displayProperty=fullName>  \\n  \\n-   <xref:System.Data.DataRow.GetParentRow%2A?displayProperty=fullName>  \\n  \\n If you call <xref:System.Data.DataRelationCollection.Add%2A>, the InvalidConstraintException may be thrown if a relation cannot be created based on the values supplied in the parameters. If you call <xref:System.Data.DataRowCollection.Clear%2A>, the InvalidConstraintException may be thrown if a <xref:System.Data.ForeignKeyConstraint> is enforced on the <xref:System.Data.DataRowCollection>. If you call <xref:System.Data.DataRow.GetParentRow%2A>, the InvalidConstraintException may be thrown if the columns have different data types or the tables do not belong to the same <xref:System.Data.DataSet>.\"\n  syntax:\n    content: 'public class InvalidConstraintException : System.Data.DataException'\n  inheritance:\n  - System.Exception\n  - System.SystemException\n  - System.Data.DataException\n  implements: []\n  inheritedMembers: []\n  platform:\n  - net462\n- uid: System.Data.InvalidConstraintException.#ctor\n  id: '#ctor'\n  parent: System.Data.InvalidConstraintException\n  langs:\n  - csharp\n  name: InvalidConstraintException()\n  nameWithType: InvalidConstraintException.InvalidConstraintException()\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException()\n  type: Constructor\n  assemblies:\n  - System.Data\n  namespace: System.Data\n  summary: Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class.\n  syntax:\n    content: public InvalidConstraintException ();\n    parameters: []\n  overload: System.Data.InvalidConstraintException.#ctor*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.InvalidConstraintException.#ctor(System.String)\n  id: '#ctor(System.String)'\n  parent: System.Data.InvalidConstraintException\n  langs:\n  - csharp\n  name: InvalidConstraintException(String)\n  nameWithType: InvalidConstraintException.InvalidConstraintException(String)\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException(String)\n  type: Constructor\n  assemblies:\n  - System.Data\n  namespace: System.Data\n  summary: Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class with the specified string.\n  syntax:\n    content: public InvalidConstraintException (string s);\n    parameters:\n    - id: s\n      type: System.String\n      description: The string to display when the exception is thrown.\n  overload: System.Data.InvalidConstraintException.#ctor*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.InvalidConstraintException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)\n  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'\n  parent: System.Data.InvalidConstraintException\n  langs:\n  - csharp\n  name: InvalidConstraintException(SerializationInfo,StreamingContext)\n  nameWithType: InvalidConstraintException.InvalidConstraintException(SerializationInfo,StreamingContext)\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException(SerializationInfo,StreamingContext)\n  type: Constructor\n  assemblies:\n  - System.Data\n  namespace: System.Data\n  summary: Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class with serialization information.\n  syntax:\n    content: protected InvalidConstraintException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);\n    parameters:\n    - id: info\n      type: System.Runtime.Serialization.SerializationInfo\n      description: The data that is required to serialize or deserialize an object.\n    - id: context\n      type: System.Runtime.Serialization.StreamingContext\n      description: Description of the source and destination of the specified serialized stream.\n  overload: System.Data.InvalidConstraintException.#ctor*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.InvalidConstraintException.#ctor(System.String,System.Exception)\n  id: '#ctor(System.String,System.Exception)'\n  parent: System.Data.InvalidConstraintException\n  langs:\n  - csharp\n  name: InvalidConstraintException(String,Exception)\n  nameWithType: InvalidConstraintException.InvalidConstraintException(String,Exception)\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException(String,Exception)\n  type: Constructor\n  assemblies:\n  - System.Data\n  namespace: System.Data\n  summary: Initializes a new instance of the <xref href=\"System.Data.InvalidConstraintException\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.\n  syntax:\n    content: public InvalidConstraintException (string message, Exception innerException);\n    parameters:\n    - id: message\n      type: System.String\n      description: The error message that explains the reason for the exception.\n    - id: innerException\n      type: System.Exception\n      description: The exception that is the cause of the current exception, or a null reference (<xref uid=\"langword_csharp_Nothing\" name=\"Nothing\" href=\"\"></xref> in Visual Basic) if no inner exception is specified.\n  overload: System.Data.InvalidConstraintException.#ctor*\n  exceptions: []\n  platform:\n  - net462\nreferences:\n- uid: System.Data.DataException\n  isExternal: false\n  name: System.Data.DataException\n- uid: System.Data.InvalidConstraintException.#ctor\n  parent: System.Data.InvalidConstraintException\n  isExternal: false\n  name: InvalidConstraintException()\n  nameWithType: InvalidConstraintException.InvalidConstraintException()\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException()\n- uid: System.Data.InvalidConstraintException.#ctor(System.String)\n  parent: System.Data.InvalidConstraintException\n  isExternal: false\n  name: InvalidConstraintException(String)\n  nameWithType: InvalidConstraintException.InvalidConstraintException(String)\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException(String)\n- uid: System.String\n  parent: System\n  isExternal: true\n  name: String\n  nameWithType: String\n  fullName: System.String\n- uid: System.Data.InvalidConstraintException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)\n  parent: System.Data.InvalidConstraintException\n  isExternal: false\n  name: InvalidConstraintException(SerializationInfo,StreamingContext)\n  nameWithType: InvalidConstraintException.InvalidConstraintException(SerializationInfo,StreamingContext)\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException(SerializationInfo,StreamingContext)\n- uid: System.Runtime.Serialization.SerializationInfo\n  parent: System.Runtime.Serialization\n  isExternal: false\n  name: SerializationInfo\n  nameWithType: SerializationInfo\n  fullName: System.Runtime.Serialization.SerializationInfo\n- uid: System.Runtime.Serialization.StreamingContext\n  parent: System.Runtime.Serialization\n  isExternal: true\n  name: StreamingContext\n  nameWithType: StreamingContext\n  fullName: System.Runtime.Serialization.StreamingContext\n- uid: System.Data.InvalidConstraintException.#ctor(System.String,System.Exception)\n  parent: System.Data.InvalidConstraintException\n  isExternal: false\n  name: InvalidConstraintException(String,Exception)\n  nameWithType: InvalidConstraintException.InvalidConstraintException(String,Exception)\n  fullName: System.Data.InvalidConstraintException.InvalidConstraintException(String,Exception)\n- uid: System.Exception\n  parent: System\n  isExternal: true\n  name: Exception\n  nameWithType: Exception\n  fullName: System.Exception\n- uid: System.Data.InvalidConstraintException.#ctor*\n  parent: System.Data.InvalidConstraintException\n  isExternal: false\n  name: InvalidConstraintException\n  nameWithType: InvalidConstraintException.InvalidConstraintException\n"}